server/dbutils.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+	"github.com/joshprzybyszewski/cribbage/server/play"
+)
+
+var (
+	errInvalidUsername error = errors.New(`invalid username`)
+)
+
+func commitOrRollback(db persistence.DB, err *error) {
+	var err2 error
+	if *err != nil {
+		err2 = db.Rollback()
+	} else {
+		err2 = db.Commit()
+	}
+	if err2 != nil {
+		log.Printf("Could not commit/rollback after %+v: %+v\n", err, err2)
+	}
+}
+
+func handleAction(_ context.Context, db persistence.DB, action model.PlayerAction) error {
+	err := db.Start()
+	if err != nil {
+		return err
+	}
+	defer commitOrRollback(db, &err)
+
+	g, err := db.GetGame(action.GameID)
+	if err != nil {
+		return err
+	}
+
+	pAPIs, err := getPlayerAPIs(db, g.Players)
+	if err != nil {
+		return err
+	}
+	err = play.HandleAction(&g, action, pAPIs)
+	if err != nil {
+		return err
+	}
+	return db.SaveGame(g)
+}
+
+func createGame(_ context.Context, db persistence.DB, pIDs []model.PlayerID) (model.Game, error) {
```

Comment:
@joshprzybyszewski I'd love to hear your thoughts on how I've implemented this. I'm now injecting the db here so the server is more testable, and the top level lambdas are now responsible for getting their own db reference, e.g.
https://github.com/joshprzybyszewski/cribbage/blob/a78e56e3359e9905dc9530801e23af81e37b9fad/server/lambdas.go#L19-L27

Reply:
I like what you've done 👍 good job